src/test/java/com/microsoft/azure/relay/TestUtil.java

```diff
@@ -23,5 +25,10 @@
             offset += array.length;
         }
         return result;
-	}
+    }
+    
+    static byte[] byteBufferToArray(ByteBuffer buffer) {
```

Comment:
>byteBufferToArray(ByteBuffer buffer) [](start = 18, length = 36)

I think `buffer1.equals(buffer2)` can replace this, since it checks for equality of the array between `position()` and `limit()` (the "remaining" segment of the buffer)